### ISSUE TITLE
mc cp fail

### ISSUE THREAD

**Issue by runner-mei** [minio/mc#2231]
## Expected behaviour
print minio is invalid bucket name on console

## Actual behaviour
mc.exe cp -r lib\minio local
 0 B / 18 B [                                                         ]   0.00%
mc.exe: <ERROR> Failed to copy `lib\minio\aaabc\aaa.txt`. XML syntax error on line 9: element <link> closed by </head>
mc.exe: <ERROR> Session safely terminated. To resume session `mc session resume MqVJhpHR`

## mc version
Version: 2017-06-15T03:38:43Z
Release-tag: RELEASE.2017-06-15T03-38-43Z
Commit-id: c32b1ea48b651125e33015ea0d1596cb377685a4

## System information
window7

**Comment by CarterMcClellan** [comment]
Can you provide me a "mc.exe config host list" output please, to further root cause analysis?

**Comment by vadmeste** [comment]
It would be also great if you run the command again with --debug flag and paste it here.

**Comment by runner-mei** [comment]
Now I'm resting, no environment. But I know why, “mc.exe cp -r lib\minio local”  will “minio” as the bucket name, send request to  "http://127.0.0.1:9000/minio", but "http://127.0.0.1:9000/minio" is gui url.

You can reproduce it yourself:
md lib
md lib\minio
echo aaaa > lib\minio\aaaa.txt
mc.exe cp -r lib\minio local

**Comment by harshavardhana** [comment]
Yep this can be reproduced @runner-mei 

```
mc: <DEBUG> GET / HTTP/1.1
Host: 10.0.0.13:9000
User-Agent: Minio (linux; amd64) minio-go/3.0.0 mc/2017-07-27T22:36:34Z
Authorization: AWS4-HMAC-SHA256 Credential=**REDACTED**/20170729/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20170729T050613Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 200 OK
Transfer-Encoding: chunked
Accept-Ranges: bytes
Content-Type: application/xml
Date: Sat, 29 Jul 2017 05:06:13 GMT
Server: Minio/DEVELOPMENT.2017-07-25T01-45-17Z (linux; amd64)
Vary: Origin
X-Amz-Bucket-Region: us-east-1
X-Amz-Request-Id: 14D5B4BC9F498720

mc: <DEBUG> Response Time:  1.848382ms

mc: <ERROR> Unable to stat `home/minio`. XML syntax error on line 9: element <link> closed by </head>
 (2) ls-main.go:102 cmd.checkListSyntax(..) Tags: [home/minio]
 (1) client-url.go:181 cmd.url2Stat(..) Tags: [home/minio]
 (0) client-s3.go:799 cmd.(*s3Client).Stat(..)
 Release-Tag:DEVELOPMENT.2017-07-27T22-36-34Z | Commit:11a89fe253c3 | Host:prtsc | OS:linux | Arch:amd64 | Lang:go1.7.5 | Mem:2.2MB/5.5MB | Heap:2.2MB/3.7MB
```

We will fix it..

**Comment by harshavardhana** [comment]
FWIW a regression was introduced on the server which leads to this error. 

```
commit e5d4e7aa9d7ba16bbf0c0fe135eb05102132b528
Author: Harshavardhana <harsha@minio.io>
Date:   Fri Mar 3 03:01:42 2017 -0800

    web: Validate if bucket names are reserved (#3841)
    
    Both '.minio.sys' and 'minio' should be never allowed
    to be created from web-ui and then fail to list it
    by filtering them out.
    
    Fixes #3840
```

**Comment by harshavardhana** [comment]
We need to open a server bug here @CarterMcClellan @vadmeste
